eclair-node/src/main/scala/fr/acinq/eclair/Plugin.scala

```diff
@@ -0,0 +1,29 @@
+package fr.acinq.eclair
+
+import java.io.File
+import java.net.{URL, URLClassLoader}
+
+import org.clapper.classutil.ClassFinder
+
+trait Plugin {
+
+  def onSetup(setup: Setup): Unit
+
+  def onKit(kit: Kit): Unit
```

Comment:
Not sure i would expose the entire `Kit`, in this way plugins enlarge the attack surface of eclair. An insecure plugin can be exploited to access channel data.

Reply:
It allows to do powerful stuff on top of eclair, and you are always taking risks by running someone else's code.